api/client/output.proto

```diff
@@ -13,17 +13,39 @@ message Counter {
 }
 
 message Percentile {
-  google.protobuf.Duration duration = 1;
+  oneof duration_type {
```

Comment:
@mum4k I would love early feedback here with respect to tracking durations and response sizes in statistics like we discussed yesterday.
This change in `output.proto` is non-breaking; it adds a new `domain` field to indicate what we are tracking, and adds a bunch of `oneof`s / extra fields for the non-duration valued fields.
I followed up all the way to the CLI output formatter to get a feel for what this approach ends up like.
If this is acceptable, to round this up, I would need to update the other output formatters as well so we wire through the newly tracked statistics in all formats we emit.

Reply:
From what I know, this is a backward compatible change. I don't think we need to add the enums though, since the oneof itself can act as such. One can figure out which one of the oneof fields was set by calling `message.has_field()`.

Adding @htuch to get his feedback on whether this is a change we want to allow on the API, but looks good from my perspective.

Reply:
From a gRPC/proto perspective, this is not a breaking wire change. It will break the Go language bindings, but that is probably OK for NIghthawk stuff today.

I'm unclear on what value the raw value is adding here.. why is `duration` not sufficient?

Reply:
> From a gRPC/proto perspective, this is not a breaking wire change. It will break the Go language bindings, but that is probably OK for NIghthawk stuff today.
> 
> I'm unclear on what value the raw value is adding here.. why is `duration` not sufficient? Generally aiming for minimalism in wire representations.

In this PR we start tracking the number of bytes observed in response headers and bodies. Using the duration-valued fields for that as a storage backend didn't seem like the right thing to do to me.

Reply:
I see. LGTM.